### PR TITLE
MiKo_2012 is now better able to fix a XAML default comment

### PR DESCRIPTION
**Pull request by RalfKoban**
- Improve XAML comment code fix by generating meaningful summaries from type names (with abbreviation expansion and article selection) instead of always using "Represents a TODO"

- Update analyzer logic for "used in" phrases to be case-sensitive

- Add tests for new summary generation behavior and edge cases

### CODE REVIEW

File: MiKo.Analyzer.Shared/Rules/Documentation/MiKo_2012_CodeFixProvider.cs [hunk 1]
```diff
@@ -134,7 +134,23 @@ private static SyntaxNode GetUpdatedSyntax(XmlElementSyntax comment, XmlTextSynt
...
             {
                 // seems like this is the default comment for WPF controls
-                return Comment(comment, XmlText("Represents a TODO"));
+                var type = comment.GetEnclosing<TypeDeclarationSyntax>();
+                var name = type.GetName();
+
+                if (name.IsNullOrEmpty())
+                {
+                    return Comment(comment, XmlText("Represents a TODO"));
+                }
```
[the remark this answers is not in the document]
Comment (RalfKoban) [reply MiKo.Analyzer.Shared/Rules/Documentation/MiKo_2012_CodeFixProvider.cs:143]: `GetName()` is prepared to receive `null` and returns an empty string in such case.

File: MiKo.Analyzer.Shared/Rules/Documentation/MiKo_2012_CodeFixProvider.cs [hunk 2]
```diff
@@ -134,7 +134,23 @@ private static SyntaxNode GetUpdatedSyntax(XmlElementSyntax comment, XmlTextSynt
...
+
+                var fixedName = AbbreviationFinder.FindAndReplaceAllAbbreviations(name);
+                var nameSpan = type.IsKind(SyntaxKind.InterfaceDeclaration) && fixedName[0] == 'I'
+                               ? fixedName.AsSpan(1) // do not use the leading 'I' of interfaces
+                               : fixedName.AsSpan();
+
+                var article = ArticleProvider.GetArticleFor(nameSpan, FirstWordAdjustment.StartLowerCase);
+                var continuation = nameSpan.WordsAsSpan().Select(_ => _.Text.ToLowerCaseAt(0)).ConcatenatedWith(" ");
+
+                return Comment(comment, XmlText($"Represents {article}{continuation}."));
```
[the remark this answers is not in the document]
Comment (RalfKoban) [reply MiKo.Analyzer.Shared/Rules/Documentation/MiKo_2012_CodeFixProvider.cs:153]: It is drastically unlikely that an interface is named exactly `I`
